pkg/cloud/gcp/provider.go

```diff
@@ -1539,25 +1552,55 @@ func (gcp *GCP) isValidPricingKey(key models.Key) bool {
 }
 
 // NodePricing returns GCP pricing data for a single node
-func (gcp *GCP) NodePricing(key models.Key) (*models.Node, error) {
+func (gcp *GCP) NodePricing(key models.Key) (*models.Node, models.PricingMetadata, error) {
+	meta := models.PricingMetadata{}
```

Comment:
This function is not materially changed. It just has extra lines adding in the metadata for the new diagnostics API.